pkg/reconciler/revision/revision.go

```diff
@@ -136,24 +138,26 @@ func (c *Reconciler) reconcileDigest(ctx context.Context, rev *v1.Revision) erro
 		digestSlice = append(digestSlice, v)
 	}
 	rev.Status.ContainerStatuses = make([]v1.ContainerStatuses, len(rev.Spec.Containers))
-	for i, container := range rev.Spec.Containers {
-		for _, v := range digestSlice {
-			if v.digestError != nil {
-				rev.Status.MarkContainerHealthyFalse(v1.ReasonContainerMissing,
-					v1.RevisionContainerMissingMessage(
-						v.image, v.digestError.Error()))
-				return v.digestError
-			}
-			if container.Name != v.containerName {
-				continue
-			}
-			if v.isServingContainer {
-				rev.Status.DeprecatedImageDigest = v.digestValue
-			}
-			rev.Status.ContainerStatuses[i] = v1.ContainerStatuses{
-				Name:        v.containerName,
-				ImageDigest: v.digestValue,
-			}
+	for _, v := range digestSlice {
```

Comment:
I also wonder if we can just join the two loops into one. I am not sure why we iterate twice.